content/news/2023-10-21-bevy-0.12/index.md

```diff
@@ -98,178 +142,60 @@ Notice how the instance data can be packed much more tightly, fitting the same a
-
-#### Assumptions
-
-The design of the automatic batching/instanced draws in Bevy makes some assumptions to enable a reasonable solution:
-
-* Only entities with prepared assets are queued to render phases
-* View bindings are constant across a render phase for a given draw function, as phases are per-view
-* `batch_and_prepare_render_phase` is the only system that performs batching and has sole responsibility for preparing per-instance (i.e. mesh uniform) data
-
-If these assumptions do not work for your use case, then you can add the `NoAutomaticBatching` component to your entities to opt-out and do your own thing. Note that mesh uniform data will still be written to the `GpuArrayBuffer` and can be used in your own mesh bind groups.
-
-#### Instanced Draw Performance
-
-We can batch draws into a single instanced draw in some situations now that per-instance mesh uniform data is in a `GpuArrayBuffer`. If the mesh entity is using the same mesh asset, and same material asset, then it can be batched!
-
-Using the same approach as 0.11 with one dynamic offset binding per mesh entity, and comparing to either storage buffers or batched uniform buffers:
-
-2D meshes: `bevymark --benchmark --waves 160 --per-wave 1000 --mode mesh2d --ordered-z` which spawns 160 waves of 1000 2D quad meshes, producing 160 instanced draws of 1000 instances per draw.
-
-![0.12-2DMeshes](0.12-2DMeshes.svg)
-<div style="font-size: 1.0rem" class="release-feature-authors">Tested on an M1 Max, limiting the BatchedUniformBuffer batch size to 1 versus how it works in 0.12.</div>
-
-3D meshes: `many_cubes` which spawn 160,000 cubes, of which ~11,700 are visible in the view. These are drawn using a single instanced draw of all visible cubes which enables up to **100% increase in frame rate (2x)**!
-
-![0.12-3DMeshes](0.12-3DMeshes.svg)
-<div style="font-size: 1.0rem" class="release-feature-authors">Tested on an M1 Max, limiting the BatchedUniformBuffer batch size to 1 versus how it works in 0.12.</div>
-
-These performance benefits can be leveraged on all platforms, including WebGL2!
-
-## Rendering Performance Improvements
-
-### EntityHashMap
+## EntityHashMap Renderer Optimization
 
 <div class="release-feature-authors">authors: Rob Swain (@superdump), @robtfm, @pcwalton, @jancespivo, @SkiFire13, @nicopap</div>
 
-#### The Performance Problem
-
-Since Bevy 0.6, Bevy's renderer has used a separate render world to store an extracted snapshot of the simulated data from the main world to enable pipelined rendering of frame N in the render app, while the main app simulates frame N+1.
+Since **Bevy 0.6**, Bevy's renderer has extracted data from the "app world" into a separate "render world". This enables [Pipelined Rendering](/news/bevy-0-6/#pipelined-rendering-extract-prepare-queue-render), which renders frame N in the render app, while the main app simulates frame N+1.
```

Comment:
@cart as far as I am aware we always call them the main world and render world. Or main app and render app, depending on whether we’re talking about the world or app specifically.

Reply:
Yup thats preferable!